### PR TITLE
Encrypt k8s at rest

### PR DESCRIPTION
**Pull request by jgsqware**
Related to https://github.com/giantswarm/giantswarm/issues/1907

This enable kubernetes encryption at rest. It use new `randomkeytpr` to retreive the encryption key created by `kubernetesd`.

### CODE REVIEW

**Comment by jgsqware** [comment]
Need this PR merge before this one https://github.com/giantswarm/kubernetesd/pull/279

**Comment by jgsqware** [comment]
PR updated, Could re-review please?

File: service/create/assets.go [hunk 1]
```diff
@@ -21,3 +24,31 @@ func (s *Service) encodeTLSAssets(assets certificatetpr.AssetsBundle, svc *kms.K
 
 	return compTLS, nil
 }
+
+func (s *Service) encodeKeyAssets(assets map[randomkeytpr.Key][]byte, svc *kms.KMS, kmsKeyArn string) (*randomkeytpr.CompactRandomKeyAssets, error) {
+
+	encryptionKey, ok := assets[randomkeytpr.EncryptionKey]
```
Comment (xh3b4sd) [inline service/create/assets.go]: This is not generic. When another key was added this has to be adapted as well. Is this on purpose?

Comment (jgsqware) [reply service/create/assets.go]: We should have this key in the map, So I was thinking to validate it later on, like we do for certificates:

WDYT?

Comment (xh3b4sd) [reply service/create/assets.go]: I think we should move forward now because we will change nearly everything when we start  using Cloudformation.

Comment (rossf7) [reply service/create/assets.go]: Checking we have the right keys makes sense to me.

**Pull request merged**